crates/rspack_plugin_javascript/src/visitors/dependency/url_scanner.rs

```diff
@@ -7,50 +7,58 @@ use swc_core::ecma::{
 };
 
 use super::expr_matcher;
+use super::worker_scanner::WorkerScanner;
 use crate::dependency::URLDependency;
+
 pub struct UrlScanner<'a> {
   pub dependencies: &'a mut Vec<Box<dyn ModuleDependency>>,
+  worker_scanner: &'a WorkerScanner<'a>,
 }
 
 // new URL("./foo.png", import.meta.url);
 impl<'a> UrlScanner<'a> {
-  pub fn new(dependencies: &'a mut Vec<Box<dyn ModuleDependency>>) -> Self {
-    Self { dependencies }
+  pub fn new(
+    dependencies: &'a mut Vec<Box<dyn ModuleDependency>>,
+    worker_scanner: &'a WorkerScanner<'a>,
+  ) -> Self {
+    Self {
+      dependencies,
+      worker_scanner,
+    }
+  }
+}
+
+pub fn match_new_url(new_expr: &NewExpr) -> Option<(u32, u32, String)> {
+  if matches!(&*new_expr.callee, Expr::Ident(Ident { sym: js_word!("URL"), .. }))
+  && let Some(args) = &new_expr.args
+  && let (Some(first), Some(second)) = (args.first(), args.get(1))
+  && let (
+    ExprOrSpread { spread: None, expr: box Expr::Lit(Lit::Str(path)) },
+    ExprOrSpread { spread: None, expr: box expr },
+  ) = (first, second) && expr_matcher::is_import_meta_url(expr) {
+    return Some((path.span.real_lo(), expr.span().real_hi(), path.value.to_string()))
```

Comment:
```suggestion
   if matches!(&*new_expr.callee, Expr::Ident(Ident {sym: js_word!("URL"),..}))
    && let Some(args) = &new_expr.args 
    && let [ExprOrSpread { spread: None, expr: box Expr::Lit(Lit::Str(path)) }, ExprOrSpread { spread: None, expr: box expr }] = &args[..]
    && expr_matcher::is_import_meta_url(expr) {
    return Some((path.span.real_lo(), expr.span().real_hi(), path.value.to_string()))
```

NIT: you can save a let chain